opentelemetry/src/main/scala/zio/telemetry/opentelemetry/metrics/internal/Instrument.scala

```diff
@@ -102,7 +102,15 @@ object Instrument {
 
             unit.foreach(builder.setUnit)
             description.foreach(builder.setDescription)
-            boundaries.foreach(seq => builder.setExplicitBucketBoundariesAdvice(seq.map(Double.box).asJava))
+            boundaries.foreach(seq =>
+              builder.setExplicitBucketBoundariesAdvice(
+                seq
+                  // ZIO uses explicit Double.MaxValue boundary as upper bound. OTEL uses internal "inf" value instead
+                  .filter(_ != Double.MaxValue)
+                  .map(Double.box)
+                  .asJava
+              )
+            )
```

Comment:
Nitpick: for the sake of syntactic consistency

```suggestion
            boundaries.foreach { seq =>
              builder.setExplicitBucketBoundariesAdvice(
                seq
                  // ZIO uses explicit Double.MaxValue boundary as upper bound. OTEL uses internal "inf" value instead
                  .filter(_ != Double.MaxValue)
                  .map(Double.box)
                  .asJava
              )
            }
```